assets/chat.js

```diff
@@ -1,30 +1,18 @@
 import Chat from "./chat/js/chat";
 import emotes from "./emotes.json";
 
-$.when(
-    new Promise(res =>
-        $.getJSON(`${API_URI}/api/chat/me`)
-            .done(res)
-            .fail(() => res(null))
-    ),
-    new Promise(res =>
-        $.getJSON(`${API_URI}/api/chat/history`)
-            .done(res)
-            .fail(() => res(null))
-    ),
-    new Promise(res =>
-        $.getJSON(`${API_URI}/api/chat/viewer-states`)
-            .done(res)
-            .fail(() => res([]))
-    )
-).then(
-    (userAndSettings, history, viewerStates) =>
-        (window.__chat__ = new Chat()
-            .withUserAndSettings(userAndSettings)
-            .withEmotes(emotes)
-            .withGui()
-            .withViewerStates(viewerStates)
-            .withHistory(history)
-            .withWhispers()
-            .connect(WEBSOCKET_URI))
-);
+const init = async v => {
+    const userInfo = async v => await new Promise(res => $.getJSON(`${API_URI}/api/chat/me`).done(res).fail(() => res(null)));
+    const history = async v => await new Promise(res => $.getJSON(`${API_URI}/api/chat/history`).done(res).fail(() => res(null)));
+    const viewerStates = async v => await new Promise(res => $.getJSON(`${API_URI}/api/chat/viewer-states`).done(res).fail(() => res(null)));
+
+    const promiseSync = await Promise.all([userInfo(), history(), viewerStates()]);
```

Comment:
Instead of loading user info, then history, etc. they are loaded all at once. Then, in the second batch user settings, whispers, are all loaded. The two need to be separate as the second set depends on the information from the first.

The dev build worked without setting ```window.__chat___``` but just in case it is set for mobile it was set again, here.